platform/src/apis/Platform.Api.Establishment/Configuration/Services.cs

```diff
@@ -49,16 +52,27 @@ internal static void Configure(IServiceCollection serviceCollection)
 
         serviceCollection
             .AddSingleton<IDatabaseFactory>(new DatabaseFactory(sqlConnString))
-            .AddSingleton<ISearchConnection<LocalAuthority>>(new SearchConnection<LocalAuthority>(searchEndpoint, searchCredential, ResourceNames.Search.Indexes.LocalAuthority))
-            .AddSingleton<ISearchConnection<School>>(new SearchConnection<School>(searchEndpoint, searchCredential, ResourceNames.Search.Indexes.School))
-            .AddSingleton<ISearchConnection<Trust>>(new SearchConnection<Trust>(searchEndpoint, searchCredential, ResourceNames.Search.Indexes.Trust))
-            .AddSingleton<ISearchConnection<ComparatorSchool>>(new SearchConnection<ComparatorSchool>(searchEndpoint, searchCredential, ResourceNames.Search.Indexes.SchoolComparators))
-            .AddSingleton<ISearchConnection<ComparatorTrust>>(new SearchConnection<ComparatorTrust>(searchEndpoint, searchCredential, ResourceNames.Search.Indexes.TrustComparators))
+            .AddSingleton<ISearchConnection<LocalAuthority>>(x => new SearchConnection<LocalAuthority>(
+                new SearchClient(searchEndpoint, ResourceNames.Search.Indexes.LocalAuthority, searchCredential),
+                x.GetService<ITelemetryService>()))
+            .AddSingleton<ISearchConnection<School>>(x => new SearchConnection<School>(
+                new SearchClient(searchEndpoint, ResourceNames.Search.Indexes.School, searchCredential),
+                x.GetService<ITelemetryService>()))
+            .AddSingleton<ISearchConnection<Trust>>(x => new SearchConnection<Trust>(
+                new SearchClient(searchEndpoint, ResourceNames.Search.Indexes.Trust, searchCredential),
```

Comment:
[Suggestion] could possibly add the SearchClient as a Keyed Singleton?


https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.servicecollectionserviceextensions.addkeyedsingleton

Reply:
Ignore me - it's not going to be that helpful

Reply:
Looks funky! I'll have a read

Reply:
&lt;sigh&gt; https://github.com/microsoft/ApplicationInsights-dotnet/issues/2879

Reply:
I've done this anyway, as it makes it a little easier to read even taking into consideration the issue above. Just finishing validation.